src/conn/conn_dhandle.c

```diff
@@ -314,8 +314,6 @@ __wt_conn_btree_open(
 	    F_ISSET(dhandle, WT_DHANDLE_EXCLUSIVE) &&
 	    !LF_ISSET(WT_DHANDLE_LOCK_ONLY));
 
-	WT_ASSERT(session, !F_ISSET(S2C(session), WT_CONN_CLOSING));
```

Comment:
What is the motivation for removing this assertion?  It has caught real bugs relating to thread management in the past...